File: python/pcats_api_example_manuscript_4.py
import pcats_api_client as pcats_api

#example 4

jobid=pcats_api.dynamicgp(datafile="../data/example4.csv", 
        stg1_outcome="L1",
        stg1_treatment="A1",
        stg1_time="Time1",
        stg1_x_explanatory="X,M",
        stg1_x_confounding="X,M",
        stg1_outcome_type="Continuous",
        stg1_tr_hte="M",   
        stg2_outcome="Y",
        stg2_treatment="A2",
        stg2_time="Time2",
        stg2_x_explanatory="X,L1,M",
        stg2_x_confounding="X,L1,M",
        stg2_outcome_type="Continuous",
        stg2_tr2_hte="M",     
        burn_num=500,
        mcmc_num=500,
        stg1_tr_type="Discrete",
        stg2_tr_type="Discrete",
        x_categorical="M",
        method="GP")

print("JobID: {}".format(jobid))

status=pcats_api.wait_for_result(jobid)

if status=="Done":
    print(pcats_api.printgp(jobid))
else:
    print("Error")
    exit()

#example 4 CATE

jobid_cate=pcats_api.dynamicgp_cate(jobid=jobid,
        x="M",
        control_tr="0,0",
        treat_tr="1,0")

print("CATE JobID: {}".format(jobid_cate))

status=pcats_api.wait_for_result(jobid_cate)

if status=="Done":
    print(pcats_api.printgp(jobid_cate))
else:
    print("Error")

File: R/pcatsAPIclientR/pcats_api_example_manuscript_4.R
library(pcatsAPIclientR)

#example 4

jobid <- pcatsAPIclientR::dynamicGP(datafile="../../data/example4.csv",
                                    stg1.outcome="L1",
                                    stg1.treatment="A1",
                                    stg1.time="Time1",
                                    stg1.x.explanatory="X,M",
                                    stg1.x.confounding="X,M",
                                    stg1.outcome.type="Continuous",
                                    stg1.tr.hte="M",   
                                    stg1.tr.type="Discrete",
                                    stg2.outcome="Y",
                                    stg2.treatment="A2",
                                    stg2.time="Time2",
                                    stg2.x.explanatory="X,L1,M",
                                    stg2.x.confounding="X,L1,M",
                                    stg2.outcome.type="Continuous", 
                                    stg2.tr2.hte="M",     
                                    stg2.tr.type="Discrete",
                                    burn.num=500,
                                    mcmc.num=500,
                                    x.categorical="M",
                                    method="GP")

cat(paste0("JobID: ",jobid,"\n"))

status <- pcatsAPIclientR::wait_for_result(jobid)

if (status=="Done") {
   cat(pcatsAPIclientR::printgp(jobid))
}

# CATE
jobidcate <- pcatsAPIclientR::dynamicGP.cate(
        jobid=jobid,
        x="M",
        control.tr="0,0",
        treat.tr="1,0")

cat(paste0("JobIDCate: ",jobidcate,"\n"))

status <- pcatsAPIclientR::wait_for_result(jobidcate)

if (status=="Done") {
  cat(pcatsAPIclientR::printgp(jobidcate))
}

